File: python3/509.斐波那契数.py
#
# @lc app=leetcode.cn id=509 lang=python3
#
# [509] 斐波那契数
#

# @lc code=start
class Solution:
    def fib(n: int) -> int:
        # 终止条件 f(1) = 0, f(2) = 1
        if n == 1 or n == 2:
            return n - 1
        # 递归调用 f(n) = f(n-1) + f(n-2)
        res = fib(n - 1) + fib(n - 2)
        # 返回结果 f(n)
        return res
        
# @lc code=end

File: c/509.斐波那契数.c
/*
 * @lc app=leetcode.cn id=509 lang=c
 *
 * [509] 斐波那契数
 */

// @lc code=start
int fib(int n) {
    // if (n <= 0) return 0; //这里是从第0位开始计算的
    // if (n == 1) return 1;
    // return fib(n - 1) + fib(n - 2);
    if(n == 0 || n == 1)
        return n;
    
    return fib(n - 1) + fib(n - 2);
}
// @lc code=end

